sdk/java/dagger-java-sdk/src/main/java/io/dagger/client/exception/DaggerExceptionUtils.java

```diff
@@ -0,0 +1,96 @@
+package io.dagger.client.exception;
+
+import static io.dagger.client.exception.DaggerExceptionConstants.CMD_KEY;
+import static io.dagger.client.exception.DaggerExceptionConstants.ENHANCED_MESSAGE;
+import static io.dagger.client.exception.DaggerExceptionConstants.EXIT_CODE_KEY;
+import static io.dagger.client.exception.DaggerExceptionConstants.FULL_MESSAGE;
+import static io.dagger.client.exception.DaggerExceptionConstants.SIMPLE_MESSAGE;
+import static io.dagger.client.exception.DaggerExceptionConstants.STDERR_KEY;
+import static io.dagger.client.exception.DaggerExceptionConstants.STDOUT_KEY;
+import static io.dagger.client.exception.DaggerExceptionConstants.TYPE_KEY;
+
+import io.smallrye.graphql.client.GraphQLError;
+import jakarta.json.JsonArray;
+import jakarta.json.JsonValue;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+
+public class DaggerExceptionUtils {
+
+  private DaggerExceptionUtils() {}
+
+  public static Object getExtensionValueByKey(GraphQLError error, String key) {
+    if (error == null || StringUtils.isBlank(key) || error.getExtensions() == null) {
+      return null;
+    }
+
+    return error.getExtensions().getOrDefault(key, null);
+  }
+
+  public static String getPath(GraphQLError error) {
+    return StringUtils.join(error.getPath(), ".");
+  }
+
+  public static String getCmd(GraphQLError error) {
+    Object cmdList = getExtensionValueByKey(error, CMD_KEY);
+    if (cmdList != null && cmdList instanceof JsonArray array) {
+      return array.stream()
+          .map(JsonValue::toString)
+          .collect(Collectors.joining(" "))
+          .replace("\"", "");
+    }
+    return null;
+  }
+
+  public static String getType(GraphQLError error) {
+    return String.valueOf(getExtensionValueByKey(error, TYPE_KEY));
+  }
+
+  public static String getExitCode(GraphQLError error) {
+    return String.valueOf(getExtensionValueByKey(error, EXIT_CODE_KEY));
+  }
```

Comment:
This should return an integer, not a string

Reply:
yes you are right, i I thought is really safe? the returned code could be not a integere value?